src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1255,17 +1255,16 @@ private BoundStatement ChainImplicitStructConstructor(MethodSymbol methodSymbol,
                 var codeGen = new CodeGen.CodeGenerator(method, block, builder, moduleBuilder, diagnosticsForThisMethod, optimizations, generateDebugInfo);
 
                 // We need to save additional debugging information for MoveNext of an async state machine.
-                var stateMachineMethod = method as SynthesizedStateMachineMethod;
-                bool isStateMachineMoveNextMethod = stateMachineMethod != null && method.Name == WellKnownMemberNames.MoveNextMethodName;
+                var stateMachineMoveNextMethod = method as SynthesizedStateMachineMoveNextMethod;
```

Comment:
This change doesn't make the code more fragile than it was before. The type name change reflects the change in type hierarchy. Please open a separate issue if you would like the type test removed from this code altogether.